File: sdk/typescript/src/api/health.ts
import type { HttpClient, RequestOptions } from "../http.js";

/**
 * Service health check.
 */
export class HealthApi {
  constructor(private readonly http: HttpClient) {}

  /** Health check endpoint. Returns server uptime and status information. */
  check<T = unknown>(options?: RequestOptions): Promise<T> {
    return this.http.get<T>("/", options);
  }
}

File: src/api/health.rs
//! Service health check.

use reqwest::Method;
use serde_json::Value;

use crate::{Error, HttpClient};

/// Typed client for the health-check route.
pub struct HealthApi<'a> {
    http: &'a HttpClient,
}

impl<'a> HealthApi<'a> {
    pub fn new(http: &'a HttpClient) -> Self {
        Self { http }
    }

    /// Health check endpoint. Returns server uptime and status information.
    pub async fn check(&self) -> Result<Value, Error> {
        self.http.send(Method::GET, "/", &[], None, true).await
    }
}
